src/gluonts/dataset/split/splitter.py

```diff
@@ -281,6 +281,8 @@ def _test_slice(
         self, item: TimeSeriesSlice, offset: int = 0
     ) -> TimeSeriesSlice:
         offset_ = self.split_offset + offset + self.prediction_length
+        if self.split_offset < 0 and offset_ >= 0:
```

Comment:
Why do you check for `offset_ >= 0:`?

Reply:
So that when `offset_ > 0` it gets increased, exceeds `len(item)` and fails the assertion that follows (which is what we want in this case).

Maybe instead of an assertion we raise a `ValueError` with a nice message?

Reply:
Or should we let the final indexing operation surface the out-of-bounds index?